include/deal.II/differentiation/ad/ad_helpers.h

```diff
@@ -709,13 +731,344 @@ namespace Differentiation
+       */
+      //@{
+
+      /**
+       * The constructor for the class.
+       *
+       * @param[in] n_independent_variables The number of independent variables
+       * that will be used in the definition of the functions that it is
+       * desired to compute the sensitivities of. In the computation of
+       * $\mathbf{f}(\mathbf{X})$, this will be the number of inputs
+       * $\mathbf{X}$, i.e. the dimension of the range space.
+       * @param[in] n_dependent_variables The number of scalar functions to be
+       * defined that will have a sensitivity to the given independent
+       * variables. In the computation of $\mathbf{f}(\mathbf{X})$, this will
+       * be the number of outputs $\mathbf{f}$, i.e. the dimension of the
+       * domain or image space.
+       */
+      ADHelperCellLevelBase(const unsigned int n_independent_variables,
+                            const unsigned int n_dependent_variables);
+
+      /**
+       * Destructor
+       */
+      virtual ~ADHelperCellLevelBase();
+
+      //@}
+
+      /**
+       * @name Independent variables
+       */
+      //@{
+
+      /**
+       * Register the complete set of independent variables $\mathbf{X}$ that
+       * represent the local degree-of-freedom values.
+       *
+       * @param[in] dof_values A vector field associated with local
+       * degree-of-freedom values on the current finite element. These define
+       * the values of all independent variables. When considering taped AD
+       * numbers with branching functions, to avoid potential issues with branch
+       * switching it may be a good idea to choose these values close to those
+       * that will be later evaluated and linearized around.
+       *
+       * @note The input value type must correspond to this class's number type.
+       * So if this class is templated on type double then the number type
+       * associated with value must also be of type double.
+       *
+       * @note For taped AD numbers, this operation is only valid in recording mode.
+       */
+      void
+      register_dof_values(const std::vector<scalar_type> &dof_values);
+
+      /**
+       * Register the complete set of independent variables $\mathbf{X}$ that
+       * represent the local degree-of-freedom values.
+       *
+       * @param[in] values A global field from which the values of all
+       * independent variables will be extracted. This typically will be the
+       * solution vector around at which point a residual vector is to be
+       * computed and around which linearization is to occur.
+       * When considering taped AD numbers with branching functions, to avoid
+       * potential issues with branch switching it may be a good idea to choose
+       * these values close to those that will be later evaluated and linearized
+       * around.
+       * @param[in] local_dof_indices A vector of degree-of-freedom indices from
+       * which to extract the local degree-of-freedom values.
```

Comment:
maybe also say here that this would then be what `cell->get_dof_indices()` would typically return, since you also give an example for the other argument